pkg/agent/openflow/pipeline.go

```diff
@@ -1411,22 +1474,28 @@ func (f *featurePodConnectivity) arpResponderStaticFlow() binding.Flow {
 
 // podIPSpoofGuardFlow generates the flow to check IP packets from local Pods. Packets from the Antrea gateway will not be
 // checked, since it might be Pod to Service connection or host namespace connection.
-func (f *featurePodConnectivity) podIPSpoofGuardFlow(ifIPs []net.IP, ifMAC net.HardwareAddr, ifOFPort uint32) []binding.Flow {
+func (f *featurePodConnectivity) podIPSpoofGuardFlow(ifIPs []net.IP, ifMAC net.HardwareAddr, ifOFPort uint32, vlanID uint16) []binding.Flow {
 	cookieID := f.cookieAllocator.Request(f.category).Raw()
 	var flows []binding.Flow
 	for _, ifIP := range ifIPs {
 		ipProtocol := getIPProtocol(ifIP)
 		targetTable := SpoofGuardTable.ofTable.GetNext()
+		ctZoneTypeRegMark := IPCtZoneTypeRegMark
 		if ipProtocol == binding.ProtocolIPv6 {
 			targetTable = IPv6Table.ofTable.GetID()
+			ctZoneTypeRegMark = IPv6CtZoneTypeRegMark
 		}
-		flows = append(flows, SpoofGuardTable.ofTable.BuildFlow(priorityNormal).
+		fb := SpoofGuardTable.ofTable.BuildFlow(priorityNormal).
 			Cookie(cookieID).
 			MatchProtocol(ipProtocol).
 			MatchInPort(ifOFPort).
 			MatchSrcMAC(ifMAC).
-			MatchSrcIP(ifIP).
-			Action().ResubmitToTables(targetTable).
+			MatchSrcIP(ifIP)
+		if f.connectUplinkToBridge {
+			fb = fb.Action().LoadRegMark(ctZoneTypeRegMark).
+				Action().LoadToRegField(VLANIDField, uint32(vlanID))
```

Comment:
As mentioned in https://github.com/antrea-io/antrea/pull/3247#discussion_r829638323 , it will break action=NORMAL.
This PR follows OVS behavior, which load VLANIDField in output stage.